main.go

```diff
@@ -29,44 +29,52 @@ func rssHandler(rw http.ResponseWriter, r *http.Request) {
 		Created:     time.Now(),
 	}
 
+	itemCh := make(chan *feeds.Item)
 	for _, feed := range sourceFeeds {
-		fetch(feed, master)
+		go fetch(feed, itemCh)
 	}
+	addItems(itemCh, master)
 
 	sort.Sort(byCreated(master.Items))
-
 	result, _ := master.ToAtom()
 	fmt.Fprintln(rw, result)
 }
 
-func fetch(feed sourceFeed, master *feeds.Feed) {
-	fetcher := rss.New(5, true, chanHandler, makeHandler(master, feed.name))
-	client := &http.Client{
-		Timeout: time.Second,
+func addItems(itemCh chan *feeds.Item, master *feeds.Feed) {
+	timeout := time.After(3 * time.Second)
```

Comment:
@croaky good catch. You're right.... without closing the `itemCh` this code always runs for 3 seconds!